### ISSUE TITLE
Modal: Tooltip is missing from the Close button on focus

### ISSUE THREAD

**Issue by thefirstartist** [carbon-design-system/carbon#14988]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

V10, V11

### React version

v. 1.40

### Automated testing tool and ruleset

n/a

### Description

When navigating with keyboard on a modal, tooltip is missing from the 'Close' button on focus. Tooltip is required for non-text content. 
<img width="1305" alt="image" src="https://github.com/carbon-design-system/carbon/assets/40870771/02d8a48b-afb3-47e1-8d94-3f1f53518a32">

Here's Carbon's accessibility guidance on buttons: 
Icon-only buttons have their labels exposed automatically on hover and focus.
https://carbondesignsystem.com/components/button/accessibility#behavior
<img width="750" alt="image" src="https://github.com/carbon-design-system/carbon/assets/40870771/a77958ee-1e31-49d9-a36c-135d85e57081">

  
Also, when using a mouse to hover the 'Close' button, the tooltip is incorrect. (This is also covered in [issue #14323](https://github.com/carbon-design-system/carbon/issues/14323))

<img width="1402" alt="image" src="https://github.com/carbon-design-system/carbon/assets/40870771/8b942664-b690-4504-8816-9788e31065e6">

### WCAG 2.1 Violation

Success Criterion 1.1.1: Non-text Content

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-modal--default&globals=theme:white

### Steps to reproduce

The violation can be found in Carbon site or storybook.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
`Modal` needs to be refactored to use `IconButton` so that it uses the correct Tooltip

**Comment by thefirstartist** [comment]
Just found out that the tooltip on the close button on notification is also missing.

<img width="1080" alt="image" src="https://github.com/carbon-design-system/carbon/assets/40870771/5be14b02-b874-4d63-b402-316320aa9754">
